src/main/java/sim/coffee/Server.java

```diff
@@ -0,0 +1,18 @@
+package sim.coffee;
+
+public class Server implements Runnable{
+    private Queue queue;
+
+    public Server(Queue queue) {
+        this.queue = queue;
+    }
+
+    public void run() {
+        while (queue.getCustomerCount() == 0) {
```

Comment:
I'm not sure I understand the purpose of this part.

I think the thread should run a loop until the queue is done, within which you get the order (automatically waits) then sleep some time based on the size of the order (simulate service).

Reply:
Do we need to have a method that returns queue size then?

Reply:
@jonleesy Yeah true, although `getCustomerCount` sounds synonymous to me (without looking at the code)